Dockerfile

```diff
@@ -1,74 +1,137 @@
-            ln -s /${PRODUCT_NAME}-${PRODUCT_VERSION}/bin/webstorm.sh /opt/run-ide.sh \
-            ;; \
-    esac && \
-    mkdir -p ${BASE_MOUNT_FOLDER}/${PRODUCT_NAME} && \
-    mkdir /etc/default/jetbrains && \
-    for f in "${BASE_MOUNT_FOLDER}" "/${PRODUCT_NAME}-${PRODUCT_VERSION}" "/etc/passwd" "/etc/default/jetbrains"; do \
-      echo "Changing permissions on ${f}" && chgrp -R 0 ${f} && \
-      chmod -R g+rwX ${f}; \
-    done && \
-    echo "idea.config.path=${BASE_MOUNT_FOLDER}/${PRODUCT_NAME}/config" > /${PRODUCT_NAME}-${PRODUCT_VERSION}/bin/idea.properties && \
-    echo "idea.system.path=${BASE_MOUNT_FOLDER}/${PRODUCT_NAME}/caches" >> /${PRODUCT_NAME}-${PRODUCT_VERSION}/bin/idea.properties && \
-    echo "idea.plugins.path=${BASE_MOUNT_FOLDER}/${PRODUCT_NAME}/plugins" >> /${PRODUCT_NAME}-${PRODUCT_VERSION}/bin/idea.properties && \
-    echo "idea.log.path=${BASE_MOUNT_FOLDER}/${PRODUCT_NAME}/logs" >> /${PRODUCT_NAME}-${PRODUCT_VERSION}/bin/idea.properties
+# Stage 2. Build Projector
+#   When buildGradle build argument is set to false, it instructs docker to consume built dist from the host.
+#   Otherwise it consumes the source code of project-server and build it inside the docker image.
+#   Requires jdk with version at least 11.
+FROM openjdk:11.0-jdk as projectorGradleBuilder
+ARG buildGradle
+ENV PROJECTOR_DIR /projector
+ADD projector-client $PROJECTOR_DIR/projector-client
+ADD projector-server $PROJECTOR_DIR/projector-server
+WORKDIR $PROJECTOR_DIR/projector-server
+RUN if [ "$buildGradle" = "true" ]; then ./gradlew clean; else echo "Skipping gradle build"; fi \
+    && if [ "$buildGradle" = "true" ]; then ./gradlew --console=plain :projector-server:distZip; else echo "Skipping gradle build"; fi
 
-# Copy fluxbox configuration
-COPY --chown=0:0 config/fluxbox /home/user/.fluxbox/init
+# Stage 3. Assemble Projector with the IDE bundle
+#   Doesn't require additional tools, there is only file operations, move, rename, chmod.
+FROM alpine:3.13.4 as projectorStaticFiles
+ENV PROJECTOR_DIR /projector
+RUN mkdir -p $PROJECTOR_DIR
+ADD jetbrains-editor-images/static $PROJECTOR_DIR
+COPY --from=ideDownloader /ide $PROJECTOR_DIR/ide
+COPY --from=projectorGradleBuilder $PROJECTOR_DIR/projector-server/projector-server/build/distributions/projector-server-1.0-SNAPSHOT.zip $PROJECTOR_DIR
+RUN unzip $PROJECTOR_DIR/projector-server-1.0-SNAPSHOT.zip \
+    && rm $PROJECTOR_DIR/projector-server-1.0-SNAPSHOT.zip \
+    && mv projector-server-1.0-SNAPSHOT $PROJECTOR_DIR/ide/projector-server \
+    && mv $PROJECTOR_DIR/ide-projector-launcher.sh $PROJECTOR_DIR/ide/bin \
+    && chmod 644 $PROJECTOR_DIR/ide/projector-server/lib/*
 
-# Copy predefined configs
-COPY --chown=0:0 config/etc /etc/
+# Stage 4. Build the main image with necessary environment for running Projector
+#   Doesn't require to be a desktop environment. Projector runs in headless mode.
+FROM fedora:32
```

Comment:
UBI8 would be better here, but if can't resolve all rpm dependencies w/o RHEL8 / pulp repos, then this is OK for an upstream base image. But please use a quay image or if that;s impossible, as least use docker.io repo prefix so it's easier to find places where we rely on Docker infrastructure.

Reply:
Every stage is now under `registry.access.redhat.com/ubi8-minimal:8.3-298` 🎉